### ISSUE TITLE
[Cygwin] multiprocessing module with pool object issue

### ISSUE THREAD

**Issue by d4fecc49-b8f9-4b81-82a9-0ab278b23cd6** [python/cpython#74748]
BPO | [30563](https://bugs.python.org/issue30563)
--- | :---
Nosy | @pitrou

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = None
closed_at = <Date 2021-06-18.16:13:05.014>
created_at = <Date 2017-06-04.06:54:26.256>
labels = ['type-bug', 'library']
title = '[Cygwin] multiprocessing module with pool object issue'
updated_at = <Date 2021-06-18.16:13:05.014>
user = 'https://bugs.python.org/JulienVerger'
```

bugs.python.org fields:
```python
activity = <Date 2021-06-18.16:13:05.014>
actor = 'iritkatriel'
assignee = 'none'
closed = True
closed_date = <Date 2021-06-18.16:13:05.014>
closer = 'iritkatriel'
components = ['Library (Lib)']
creation = <Date 2017-06-04.06:54:26.256>
creator = 'Julien Verger'
dependencies = []
files = []
hgrepos = []
issue_num = 30563
keywords = []
message_count = 5.0
messages = ['295114', '295869', '296009', '323466', '323469']
nosy_count = 3.0
nosy_names = ['pitrou', 'Julien Verger', 'nkjensen']
pr_nums = []
priority = 'normal'
resolution = 'third party'
stage = 'resolved'
status = 'closed'
superseder = None
type = 'behavior'
url = 'https://bugs.python.org/issue30563'
versions = ['Python 2.7']
```

</p></details>

**Comment by d4fecc49-b8f9-4b81-82a9-0ab278b23cd6** [comment]
Hello,

i'm trying to use the sample code that is documented here: https://docs.python.org/2/library/multiprocessing.html
from multiprocessing import Pool

```py
def f(x):
    return x*x

if __name__ == '__main__':
    p = Pool(5)
    print(p.map(f, [1, 2, 3]))
```


This code behaves differently depending on the python version that i use.
I use 2 versions of python interpreter, first on my windows 7 OS and second in cygwin.
The code behaves normaly when i execute it on windows python version, but nothing seems to happen when i use python interpreter from cygwin.
Other objects than 'pool' seems to be impacted but this short exemple helps me to show how to reproduce quickly my issue.

Here are somme informations on the python versions that i'm using:
On the windows side:
Python 2.7.13 (v2.7.13:a06454b1afa1, Dec 17 2016, 20:53:40) [MSC v.1500 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.                          

On the cygwin side:
Python 2.7.13 (default, Mar 13 2017, 20:56:15)                           
[GCC 5.4.0] on cygwin                                                    
Type "help", "copyright", "credits" or "license" for more information.

Please help

Regards

**Comment by pitrou** [comment]
How did you get Python for Cygwin? Did you compile it yourself?
Python is not supported on Cygwin (we recommend using the native Windows builds instead) and I'm not surprised that multiprocessing, which uses delicate platform-specific code, would fail working on that platform.

**Comment by d4fecc49-b8f9-4b81-82a9-0ab278b23cd6** [comment]
Hello Antoine,

I've installed python throught the cygwin.exe installer.
I've tried to add many pythons libs without more success.
I didn't know that cygwin was an unsupported platform.
Unfortunately, my goal is to develop an automation tool based on pexpect package which does not work the same on windows build (e.g. pexpect.spawn instruction wich is not compatible on windows).
I thought that i could bypass this by using pexpect on cygwin that works great but i would like to launch multiple pexpect thread in parallel by using multithread package in my cygwin environment.

Julien
